thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js

```diff
@@ -375,6 +381,87 @@ export default Ember.Controller.extend({
   },
 
   /**
+   * Get top 5 dimensions for graph
+   * @method getTopDimensions
+   * @param {Object} config - the current graph properties
+   * @return {undefined}
+   */
+  getTopDimensions(config) {
+    // TODO: Add filters in url
+    const url = `/rootcause/query?framework=relatedDimensions&anomalyStart=${config.currentStart}&anomalyEnd=${config.currentEnd}&baselineStart=${config.baselineStart}&baselineEnd=${config.baselineEnd}&analysisStart=${config.currentStart}&analysisEnd=${config.currentEnd}&urns=thirdeye:metric:${config.id}`;
+    const maxSize = 5;
+    const topDimensionGroups = [];
+
+    return fetch(url)
+      .then(checkStatus)
+      .then((dimensions) => {
+        // get top 5
+        const topDimensions = dimensions.sortBy('score').reverse().slice(0, maxSize);
+        const topDimensionLabels = [...new Set(topDimensions.map(key => key.label))];
+        // do another fetch for each dimension's group/value
+        this.fetchGraphDataForTopDimensions(topDimensionLabels, config, maxSize).then((topDimensionData) => {
+          // Set unique array of objects containing graph data for each top dimension
+          this.setProperties({
+            isFetchingDimensions: false,
+            topDimensions: topDimensionData
+          });
+        });
+      });
+  },
+
+  /**
+   * Enriches the list of functions by Id, adding the properties we may want to display.
+   * We are preparing to display the alerts that belong to the currently selected config group.
+   * @method fetchGraphDataForTopDimensions
+   * @param {Object} topDimensionLabels - the highest scoring dimensions (ex: ['continent=Europe'])
+   * @param {Object} config - the current graph properties
+   * @param {Number} targetSize - the desired number of dimensions to process
+   * @return {Ember.RSVP.Promise} A new list of dimensions
+   */
+  fetchGraphDataForTopDimensions(topDimensionLabels, config, targetSize) {
+    const dimensionList = [];
+    let dimensionObj = {};
+    let subDimensionObj = {};
+    let dimensionUrl = '';
+    let label = '';
+
+    return new Ember.RSVP.Promise((resolve) => {
+      /**
+       * Recursive iterator function walks through the topDimensionLabels array
+       * after each fetch callback
+       */
+      const buildDimensionObject = (index) => {
+        label = topDimensionLabels[index].split('=');
```

Comment:
nit: deconstructing might be cleaner here `const [ group, subDimension ] = label`
then just pass 
```
{ 
  name: label.join(':'),
  group,
  subdimension
}
```